src/main/java/be/seeseemelk/mockbukkit/ServerMock.java

```diff
@@ -2529,8 +2522,11 @@ public int getTicksPerSpawns(@NotNull SpawnCategory spawnCategory)
 	@Override
 	public int getSpawnLimit(@NotNull SpawnCategory spawnCategory)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		Preconditions.checkArgument(spawnCategory != null, "SpawnCategory cannot be null");
+		Preconditions.checkArgument(spawnCategory != SpawnCategory.MISC,
+				"SpawnCategory.%s are not supported", spawnCategory);
+
+		return this.serverConfiguration.getSpawnLimits().getOrDefault(spawnCategory, -1);
```

Comment:
```suggestion
		if(!this.serverConfiguration.getSpawnLimits().containsKey(spawnCategory)){
			throw new UnimplementedOperationException();
		}
		return this.serverConfiguration.getSpawnLimits().getOrDefault(spawnCategory, -1);
```
Update for comment https://github.com/MockBukkit/MockBukkit/pull/909/files#r1410812763